### PR TITLE
chore: Enable GitHub workflows on release branches

### PR DESCRIPTION
**Pull request by iam-veeramalla**
**What type of PR is this?**
> /kind enhancement

**What does this PR do / why we need it**:
Enable Github workflows for release branches

### CODE REVIEW

File: .github/workflows/ci-build.yaml
Comment (jopit) [inline .github/workflows/ci-build.yaml:6]: Don't we still need the workflows enabled for `vx.x.x` branches?  When we release version 0.1.0 of the operator, isn't its branch going to be `v0.1.0`, not `rhos-v0.1.0`?

**Review by jopit — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-build.yaml
Comment (jannfis) [reply .github/workflows/ci-build.yaml:6]: I think we may consider naming our release branches `release-X.Y`. IMHO, we don't need release branches for vX.Y.Z, only for vX.Y (vX.Y.Z will then be a tag within that release branch, because it is not supposed to change anymore).

Also, all this could be simplified using a glob match pattern instead of regular expressions:

* `release-*`
* `rhos-*`

Comment (iam-veeramalla) [reply .github/workflows/ci-build.yaml:6]: sure, let me change that.

Comment (iam-veeramalla) [reply .github/workflows/ci-build.yaml:6]: Made changes accordingly, Please verify if it looks good now.
@jannfis @jopit

**Review by jannfis — APPROVED** [review]
LGTM, thanks!

Please wait for @jopit's approval before merging, tho.

**Review by jopit — APPROVED** [review]

**Pull request merged**
